code/data_structures/src/tree/binary_tree/avl_tree/AVL.c

```diff
@@ -0,0 +1,332 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct node
+{
+	struct node *lcptr;
+	int data;
+	int height;
+	struct node *rcptr;
+};
+
+struct node *rptr = NULL;
+int c = 0,f = 0;
+
+int
+height(struct node *rptr)
+{
+	if (rptr == NULL)
+		return (-1);
+	else
+		return (rptr->height);
+}
+
+void
+heightUpdate(struct node *rptr)
+{
+	int leftHeight = height(rptr->lcptr);
+	int rightHeight = height(rptr->rcptr);
+
+	if (leftHeight > rightHeight)
+		rptr->height = leftHeight + 1;
+	else
+		rptr->height = rightHeight + 1;
+}
+
+int
+getBalance(struct node *rptr)
+{
+	if (rptr == NULL)
+		return (0);
+	else
+		return (height(rptr->lcptr) - height(rptr->rcptr));
+}
+
+struct node *
+RightRotate(struct node *rptr)
+{
+	struct node *jptr = rptr;
+	struct node *kptr = rptr->lcptr;
+	jptr->lcptr = kptr->rcptr;
+	kptr->rcptr = jptr;
+
+	heightUpdate(jptr);
+	heightUpdate(kptr);
+
+	return (kptr);
+}
+
+struct node *
+LeftRotate(struct node *rptr)
+{
+	struct node *jptr = srptr;
+	struct node *kptr = rptr->rcptr;
+	jptr->rcptr = kptr->lcptr;
+	kptr->lcptr = jptr;
+
+	heightUpdate(jptr);
+	heightUpdate(kptr);
+
+	return (kptr);
+}
+
+struct node *
+LeftRightRotate(struct node *rptr)
+{
+	struct node *jptr = rptr;
+	struct node *kptr = rptr->lcptr;
+	jptr->lcptr = LeftRotate(kptr);
+
+	return RightRotate(jptr);
+}
+
+struct node *
+RightLeftRotate(struct node *rptr)
+{
+	struct node *jptr = rptr;
+	struct node *kptr = rptr->rcptr;
+	jptr->rcptr = RightRotate(kptr);
+
+	return LeftRotate(jptr);
+}
+
+struct node *
+insert(struct node *rptr,int x)
```

Comment:
:sparkles: Looks better.  
Put one space between function args.

Reply:
Yeah,done. Does it look good now?